scpdt/impl.py

```diff
@@ -375,10 +377,19 @@ class DebugDTRunner(DTRunner):
     Almost verbatim copy of `doctest.DebugRunner`.
     """
     def run(self, test, compileflags=None, out=None, clear_globs=True):
-        r = super().run(test, compileflags, out, False)
-        if clear_globs:
-            test.globs.clear()
-        return r
+        """
+        Run tests in context managers:
+        np_errstate: restores the numpy errstate and printoptions when done
+        user_context_manager: allows user to plug in their own context manager
+        matplotlib_make_nongui: temporarily make the matplotlib backend non-GUI
+        """
+        with np_errstate():
+            with self.config.user_context_mgr(test):
+                with matplotlib_make_nongui():
+                    r = super().run(test, compileflags=compileflags, out=out, clear_globs=False)
+                    if clear_globs:
+                        test.globs.clear()
+                    return r
```

Comment:
To summarize our discussion offline: 
- either this context setting should happen in the `DTRunner` + inherited by `DebugDTRunner` --- then remove it from `testmod` and `testfile`;
- or this context manipulation belongs to the frontend and thus needs to be done in the plugin. 

Since the plugin seems to need to provide a specialized runner because of `out` being a list not a callable, it seems to make sense to move this logic in this runner.